src/VCR/Videorecorder.php

```diff
@@ -214,4 +226,19 @@ public function __destruct()
             $this->turnOff();
         }
     }
+
+    protected function iterateIndex(Request $request): int
+    {
+        $hash = $request->getHash();
```

Comment:
The issue with this method is that it assumes a request is unique based on its hash, but this may not always be the case.

For instance, if someone turns off the header request matcher, requests with the same body, URL, and other parameters but different headers should still be treated as the same request and receive the sequential index.

However, this solution would result in different hashes for those identical requests due to the differing headers, and both would get the same index.

I hope this is clear enough.

Would you accept PR or does this help you to resolve it?